src/Shared/AssemblyNameExtension.cs

```diff
@@ -958,5 +1008,30 @@ internal bool PartialNameCompare(AssemblyNameExtension that, PartialComparisonFl
             }
             return true;
         }
+
+        public void GetObjectData(SerializationInfo info, StreamingContext context)
+        {
+            if (asAssemblyName != null)
+            {
+                info.AddValue("asAssemblyName", true);
```

Comment:
It probably doesn't matter a lot but these could just be something like "AN" not "asAssemblyName" which adds up, with 100 assembly names that's about 15KB of extra writes, 30KB if it's written out as UTF16 (not sure).

For similar reasons you might consider "interning" some well known values like the 3 or 4 common PK and PKT almost all our assemblies have -- writing out a marker for those instead would save a lot of bytes if you write out many of these assembly names.